docker/models/containers.py

```diff
@@ -680,10 +680,34 @@ def run(self, image, command=None, stdout=True, stderr=False,
                   This mode is incompatible with ``ports``.
 
                 Incompatible with ``network``.
-            network_driver_opt (dict): A dictionary of options to provide
-                to the network driver. Defaults to ``None``. Used in
-                conjuction with ``network``. Incompatible
-                with ``network_mode``.
+            network_config (dict): A dictionary containing options that are
```

Comment:
There is an `EndpointConfig` type: https://github.com/docker/docker-py/blob/c38656dc7894363f32317affecc3e4279e1163f8/docker/types/networks.py#L5-L8

Newer versions of the Moby API (will) allow defining multiple networks on create, so let's keep this the same here, e.g. `Dict[str, EndpointConfig]`, so that we don't have to update this again / make another breaking change.

(I'm ok with this breaking change to replace `network_driver_opt` since it was just introduced in v6, so it is hopefully not that entrenched in codebases yet.)